File: lang/go/src/vac.fun/algorithms/number_theory/FibonacciNumber/FibonacciNumber.go
package FibonacciNumber

func IterativeProcedure(n int64) int64 {
    sign := int64(1)

    if n < 0 {
        if (n & 1) == 0 {
            sign = -1
        }
        n = -n
    }

    if n < 2 {
        return n
    }

    prev := int64(0)
    curr := int64(1)

    for i := int64(2); i <= n; i++ {
        prev, curr = curr, prev + curr
    }

    return sign * curr
}

func RecursiveProcedure(n int64) int64 {
    if n < 0 {
        if (n & 1) == 0 {
            return -RecursiveProcedure(-n)
        }
        return RecursiveProcedure(-n)
    }
    if n < 2 {
        return n
    }
    return RecursiveProcedure(n - 2) + RecursiveProcedure(n - 1)
}

File: archive/php/muse/algorithms/number_theory/FibonacciNumber.php
<?php

declare(strict_types=1);

namespace muse\algorithms\number_theory\FibonacciNumber;

function iterativeProcedure(int $n): int
{
    $sign = 1;

    if ($n < 0) {
        if (($n & 1) == 0) {
            $sign = -1;
        }
        $n = -$n;
    }

    if ($n < 2) {
        return $n;
    }

    $prev = 0;
    $curr = 1;

    $next = 0;
    for ($i = 2; $i <= $n; $i++) {
        $next = $prev + $curr;
        $prev = $curr;
        $curr = $next;
    }

    return $sign * $curr;
}

function recursiveProcedure(int $n): int
{
    if ($n < 0) {
        if (($n & 1) == 0) {
            return -recursiveProcedure(-$n);
        }
        return recursiveProcedure(-$n);
    }
    if ($n < 2) {
        return $n;
    }
    return recursiveProcedure($n - 2) + recursiveProcedure($n - 1);
}
